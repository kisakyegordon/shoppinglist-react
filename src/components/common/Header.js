import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';

class Header extends Component{


    isAuthenticated() {
        const token = localStorage.getItem('token');
        return token && token.length > 10;
    }
    render (){
        let logged = false;

        const guestLinks = (
            <ul className="nav navbar-nav navbar-right">
                <li> <NavLink to="/" activeClassName="active"> Home </NavLink> </li>
                <li> <NavLink to="/login" activeClassName="active"> Login </NavLink> </li>
                <li> <NavLink to="/register" activeClassName="active"> Register </NavLink></li> 
            </ul>
        );

        const userLinks = (
            <ul className="nav navbar-nav navbar-right">
                <li> <NavLink to="/" activeClassName="active"> Home </NavLink> </li>
                <li> <NavLink to="/shoppinglists" activeClassName="active"> Dashboard </NavLink> </li>
                <li> <NavLink to="/logout" activeClassName="active"> Logout </NavLink> </li>
                {/* <li> <NavLink to="/reset" activeClassName="active"> Reset Password </NavLink> </li> */}
            </ul>
        );

        return (

            
            <div>
                <div>
                    <div className="top">
                    <h2>My Shopping List</h2>
                    </div>

                    <nav className="navbar navbar-default navbar-fixed-top topnav" role="navigation">
                    <div className="container topnav">

                    <div className="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
                    {
                        this.isAuthenticated() ? userLinks : guestLinks
                    }
                        

                    </div>
                    </div>
                    </nav>
                    </div>
            </div>
        );
    }
}

export default Header;