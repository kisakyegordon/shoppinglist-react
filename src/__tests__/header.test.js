import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import Header from '../components/header/header';

describe('<Header />', () => {
    it('match a snapshot', () => {
        const wrapper = shallow(<Header />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

});