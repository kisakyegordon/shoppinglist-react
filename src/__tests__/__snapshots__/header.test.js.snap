// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> match a snapshot 1`] = `
<div>
  <div>
    <div
      className="top"
    >
      <h2>
        My Shopping List
      </h2>
    </div>
    <nav
      className="navbar navbar-default navbar-fixed-top topnav"
      role="navigation"
    >
      <div
        className="container topnav"
      >
        <div
          className="collapse navbar-collapse"
          id="bs-example-navbar-collapse-1"
        >
          <ul
            className="nav navbar-nav navbar-right"
          >
            <li>
               
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/"
              >
                 Home 
              </NavLink>
               
            </li>
            <li>
               
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/login"
              >
                 Login 
              </NavLink>
               
            </li>
            <li>
               
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                to="/register"
              >
                 Register 
              </NavLink>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>
</div>
`;
